packages/suite/src/components/suite/CoinjoinStatusBar.tsx

```diff
@@ -0,0 +1,173 @@
+    const devices = useSelector(state => state.devices);
+    const accounts = useSelector(state => state.wallet.accounts);
+    const selectedDevice = useSelector(state => state.suite.device);
+    const routerParams = useSelector(state => state.router.params);
+    const discovery = useSelector(state => state.wallet.discovery);
+
+    const { goto, selectDevice } = useActions({
+        goto: routerActions.goto,
+        selectDevice: suiteActions.selectDevice,
+    });
+
+    const relatedAccount = accounts?.find(account => account?.key === accountKey);
+
+    if (!relatedAccount) {
+        return null;
+    }
+
+    const { symbol, index, accountType, deviceState } = relatedAccount;
+
+    const isOnSelectedDevice = selectedDevice?.state === deviceState;
+    const relatedDevice = devices.find(device => device.state === deviceState);
+
+    const handleViewAccount = () => {
+        if (!isOnSelectedDevice) {
+            selectDevice(relatedDevice);
+        }
+
+        goto('wallet-index', {
+            params: {
+                symbol,
+                accountIndex: index,
+                accountType,
+            },
+        });
+    };
+
+    const { signedRounds, maxRounds, deadline } = session;
+
+    const phase = MockCoinjoinPhase.Ongoing; // TEMPORARY
+    const progress = signedRounds.length / (maxRounds / 100);
+
+    const {
+        symbol: symbolParam,
+        accountIndex: indexParam,
+        accountType: accountTypeParam,
+    } = (routerParams as WalletParams) || {};
+
+    const isOnAccountPage =
+        symbolParam === symbol && indexParam === index && accountTypeParam === accountType;
+
+    const areDevicesDiscovered = devices.every(({ state }) =>
+        discovery.find(
+            discoveryState =>
+                discoveryState.deviceState === state &&
+                discoveryState.status === DiscoveryStatus.COMPLETED,
+        ),
+    );
+
+    return (
+        <Container>
+            <ProgressPie progress={progress} />
+
+            <StatusText>
+                <Translation id={PHASE_MESSAGES[phase]} />
+
+                <Separator>•</Separator>
+
+                <Translation
+                    id="TR_COINJOIN_COUNTDOWN"
+                    values={{ rounds: maxRounds - signedRounds.length }}
+                />
+            </StatusText>
+
+            {session?.deadline && (
+                <Note>
+                    <Separator>•</Separator>
+
+                    <Translation
+                        id="TR_COINJOIN_ROUND_COUNTDOWN"
+                        values={{
+                            time: <CountdownTimer deadline={Number(deadline)} />,
+                        }}
+                    />
+                </Note>
+            )}
+
+            {!isSingle && (
+                <Note>
+                    <Separator>•</Separator>
+                    <span>{relatedDevice?.label}</span>
```

Comment:
I would expect here something different than device label (some wallet label or name, but we don't have such property, I guess), but we could do it as a followup.

Reply:
Yes, I forgot that it is per wallet, so there should be a wallet label fca2a5de5aa42220238e47af6af56bb3ed06f6ab